webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -398,7 +404,8 @@ trait NodeInfoServiceCached extends NodeInfoService with NamedZioLogger with Cac
     }
 
     //actual logic that check what to do (invalidate cache or not). We want it to be fully synchronized.
-    semaphore.flatMap(_.withPermit(
+    for {
```

Comment:
you don't need that for anymore. It should not impact perf, but code will be clearer without it.